File: codewar/V1/Day-120/evenOdd.py


def even_or_odd(number):
    """we have one number decide is it odd or even

    Args:
        number ([number]): number
    """
    if number & 1:
        return "Odd"
    else:
        return "Even"


# print(type(even_or_odd(123)))
# print(even_or_odd(22))

# ------------------------------ other solution ------------------------------ #

def second(number):
    # return "Even" if number % 2 == 0 else "Odd"
    # or
    return "Odd" if number & 1 == 1 else "Even"


print(second(123))

File: codewar/8-kyu/even_odd.cpp
#include <iostream>

using namespace std;

string even_or_odd(int number){
  // if (number % 2 == 0){
  //   return "Even";
  // } else {
  //   return "Odd";
  // }
  return number % 2 == 0 ? "Even" : "Odd";
}
int main() {
  cout << even_or_odd(2) << endl;

  return 0;
}
